docs/src/main/tut/kleisli.md

```diff
@@ -0,0 +1,273 @@
+---
+layout: default
+title:  "Kleisli"
+section: "data"
+source: "https://github.com/non/cats/blob/master/core/src/main/scala/cats/data/Kleisli.scala"
+scaladoc: "#cats.data.Kleisli"
+---
+# Kleisli
+Kleisli is a data type that will come in handy often, especially if you are working with monadic functions.
+Monadic functions are functions that return a monadic value - for instance, a function may return an
+`Option[Int]` or an `Xor[String, List[Double]]`.
+
+How then do we compose these functions together nicely? We cannot use the usual `compose` or `andThen` methods
+without having functions take an `Option` or `Xor` as a parameter, which can be strange and unwieldy.
+
+We may also have several functions which depend on some environment and want a nice way to compose these functions
+to ensure they all receive the same environment. Or perhaps we have functions which depend on their own "local"
+configuration and all the configurations together make up a "global" application configuration. How do we
+have these functions play nice with each other despite each only knowing about their own local requirements?
+
+These situations are where `Kleisli` is immensely helpful.
+
+## Functions
+One of the most useful properties of functions is that they **compose**. That is, given a function
+`A => B` and a function `B => C`, we can combine them to create a new function `A => C`. It is through
+this compositional property that we are able to write many small functions and compose them together
+to create a larger one that suits our needs.
+
+Sometimes, our functions will need to return monadic values. For instance, consider the following set of functions.
+
+```tut
+val parse: String => Option[Int] = s =>
+  try {
+    Some(s.toInt)
+  } catch {
+    case _: NumberFormatException => None
+  }
+
+val reciprocal: Int => Option[Double] =
+  i => if (i != 0) Some(1.0 / i) else None
+```
+
+As it stands we cannot use `Function1.compose` (or `Function1.andThen`) to compose these two functions.
+The output type of `parse` is `Option[Int]` whereas the input type of `reciprocal` is `Int`.
+
+This is where `Kleisli` comes into play.
+
+## Kleisli
+At it's core, `Kleisli[F[_], A, B]` is just a wrapper around the function `A => F[B]`. Depending on the
+properties of the `F[_]`, we can do different things with `Kleisli`s. For instance, if `F[_]` has a
+`FlatMap[F]` instance (we can call `flatMap` on `F[A]` values), we can
+compose two `Kleisli`s much like we can two functions.
+
+```tut
+import cats.FlatMap
+
+final case class Kleisli[F[_], A, B](run: A => F[B]) {
+  def compose[Z](k: Kleisli[F, Z, A])(implicit F: FlatMap[F]): Kleisli[F, Z, B] =
+    Kleisli[F, Z, B](z => F.flatMap(k.run(z))(run))
+}
```

Comment:
I suggest using `import cats.implicits._` here and then using syntax, rather than referring to `F`.flatMap`, e.g.

```scala
Kleisli[F, Z, B](z => k.run(z).flatMap(run))
```